zerver/management/commands/merge_streams.py

```diff
@@ -79,7 +82,13 @@ def handle(self, *args: Any, **options: str) -> None:
                 [stream_to_destroy],
                 acting_user=None,
             )
-        do_deactivate_stream(stream_to_destroy, acting_user=None)
         if len(users_to_activate) > 0:
             print(f"Adding {len(users_to_activate)} subscriptions")
             bulk_add_subscriptions(realm, [stream_to_keep], users_to_activate, acting_user=None)
+        # note: this might fail in the obscure case that the stream is used for reporting messages.
+        # in admin settings, either disable that feature or change the reporting-stream and then
+        # manually delete it.
+        try:
+            do_deactivate_stream(stream_to_destroy, acting_user=None)
```

Comment:
What's the thinking behind the reordering here -- is the point to make sure that if an error happens, we don't lose the state of which users were subscribed? 

It might be better to just have a prep commit that moves the `users_to_activate` step to be first, before the deactivation -- that certainly feels like a more rational structure for this file.